### PR TITLE
added all necessary functions

### PR DESCRIPTION
**Pull request by stino-x**
add, delete, edit, refresh(bonus)
the aforementioned are the functionality the to do list has gained in this branch the ability to add delete edit and refresh

### CODE REVIEW

**Review by franclobo — COMMENTED** [review]
Hi @stino-x, 

terminator

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have a descriptive PR, please 🙏🏽 add a descriptive summary of the changes in the current branch.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/index.js [hunk 1]
```diff
@@ -1,33 +1,242 @@
...
+// refresh function
+// const reFreshPage = (e) => {
+//   if (e.target.classList.contains('refresh')) {
+//     reFresh.setAttribute('animation', 'spin');
+//     window.location.reload();
+//     window.addEventListener('load', () => {
+//       reFresh.setTimeout(() => reFresh.classList.remove('animation'), 4500);
+//     });
+//   }
+// };
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] Please kindly remove all commented code from your codebase as it does not follow best practices and corrupts the codebase.

File: src/index.js [hunk 2]
```diff
@@ -1,33 +1,242 @@
...
+    <i class="fa-solid fa-ellipsis-vertical"></i></span>
+`;
+    parent.classList.add('listitem');
+    parent.classList.remove('edit');
+  }
+};
+activitySection.addEventListener('keypress', EditTask);
+// MyConstructor();
+// AddtoLocalStorage();
+// saveTaskstoLocalStorage(); // AddToScreen();
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] Please kindly remove all commented code from your codebase as it does not follow best practices and corrupts the codebase.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @stino-x , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/index.js [hunk 3]
```diff
@@ -1,33 +1,230 @@
+// import { cLearALL, handleCheckboxchange } from './Modules/checkbox.js';
+// import DeleteTask from './Modules/Delete.js';
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] As pointed out earlier, you should remove all commented code in all your codebase but there are still some comments

File: src/index.js [hunk 4]
```diff
@@ -1,33 +1,230 @@
...
 import './style.css';
+import reFreshPage from './Modules/refresh.js';
+// import { AddtoLocalStorage, saveTaskstoLocalStorage } from './Modules/AddtoLocalStorage.js';
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] Kindly remove this too

File: src/index.js [hunk 5]
```diff
@@ -1,33 +1,230 @@
...
+    // Create the edit form HTML
+    const span = document.createElement('span');
+    span.classList.add('activity');
+    const input = document.createElement('input');
+    input.focus();
+    input.type = 'text';
+    input.classList.add('input-edit');
+    input.value = activityArray[taskId].description;
+    const trashIcon = document.createElement('box-icon');
+    // trashIcon.addEventListener('click', deLete); // event listener
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] Kindly remove this too

File: src/index.js [hunk 6]
```diff
@@ -1,33 +1,230 @@
...
+  }
+});
+
+// deleteing a task
+
+const DeleteTask = (e) => {
+  if (e.target.classList.contains('trash')) {
+    const listItem = e.target.parentNode.parentNode;
+    const taskIndex = listItem.getAttribute('id');
+    // const taskIndex = Array.from(listItem.parentNode.children).indexOf(listItem) - 1;
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] Kindly remove this too

File: src/index.js [hunk 7]
```diff
@@ -1,33 +1,230 @@
...
+
+document.addEventListener('click', reFreshPage);
+
+// editing text in task
+const EditTask = (e) => {
+  const parent = e.target.parentNode.parentNode;
+  const index = parent.getAttribute('id');
+  if (e.key === 'Enter' && e.target.classList.contains('input-edit')) {
+    const newDescription = e.target.value;
+    // editTask(newDescription, index);
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] Kindly remove this too

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi, 
Good job implementing some of the changes requested by the previous reviewer.

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by AnselemOdims — APPROVED** [review]
### Hi  @stino-x 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @stino-x, 👋 👋  

Your project is complete! Great job for following the project requirement🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




### To Highlight :1st_place_medal: 

- Descriptive Pull Request title & summary. ✔️  
- Add and Remove functionality works Perfectly ✔️ 
- Edit Task functionality works fine ✔️ 
- Good use of GitHub flow ✔️ 

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**